lib/wibox/widget/calendar.lua

```diff
@@ -139,8 +139,36 @@ end
 -- @tparam number|nil date.day Date day
 -- @treturn widget Grid layout
 local function create_month(props, date)
-    local num_rows    = 8
-    local num_columns = props.week_numbers and 8 or 7
+    local start_row    = 3
+    local week_start   = props.start_sunday and 1 or 2
+    local last_day     = os.date("*t", os.time{year=date.year, month=date.month+1, day=0})
+    local month_days   = last_day.day
+    local column_fday  = (last_day.wday - month_days + 1 - week_start ) % 7
+
+    local num_columns  = props.week_numbers and 8 or 7
+    local start_column = num_columns - 6
+
+    -- Compute number of rows
+    -- There are at least 4 weeks in a month
+    local num_rows     = 4
+    -- On every month but february on non bisextile years
+    if last_day.day > 28 then
+        -- The number of days span over at least 5 weeks
+        num_rows = num_rows + 1
+
+        -- On month with 30+ days add 1 week if:
+        -- - if 30 days and the first day is a sunday
+        -- - if 31 days and the first days is at least saturday
+        if column_fday >= 6 then
+            if last_day.day == 30 and column_fday == 7 or last_day.day == 31 then
+                num_rows = num_rows + 1
+            end
+        end
+    -- If the first day of february is anything but a monday
```

Comment:
does it scale for the case if in system locale week starts with sunday instead of monday?

Reply:
It appears to ignore the locale settings. Please correct me if I'm wrong. Maybe the comment is not the most accurate here.

Reply:
`os.date` is not locale aware, as far as I can tell.
<https://www.lua.org/pil/22.1.html> sounds like it is fixed, and a quick test confirms that:
```shell
LC_TIME=de_DE.UTF-8 lua5.3 -e 'print(os.date("*t").wday)'
4 # would be 3, if locale aware
```

 So `wday == 1` is always Sunday.